### ISSUE TITLE
API: extend resample to handle non-datetime numeric axes?

### ISSUE THREAD

**Issue by shoyer** [pydata/pandas#12828]
Numeric indexes are pretty common with scientific datasets and [xarray](http://xarray.pydata.org). For such data, resampling to a new resolution is a pretty common operation.

Currently, resampling only works for dates. Any thoughts on making it work for all numeric axes? I think every argument in the [signature](http://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.resample.html) would still make sense for numbers, except the rule would be expected to be a number, not a string.

I expect the implementation would be relatively straightforward, given that we don't need to handle the complexity of datetime frequencies.

xref https://github.com/pydata/xarray/pull/818

**Comment by jreback** [comment]
yes, this is not very hard, basically just need a way to snap points to new points, which is what ``cut`` does (and you can generically map as well). The new ``.resample`` is pretty setup to do this as well. For completeness, can you show a simple motiviating example, with a demo-frame and output.
